node/src/accountant/scanners/mod.rs

```diff
@@ -3367,38 +4413,270 @@ mod tests {
+            err.log_error(&logger, ScanType::Payables, false);
+
+            test_log_handler.exists_log_containing(&expected_log_msg);
+        });
+    }
+
+    #[test]
+    fn log_error_works_fine_for_externally_triggered_scanning() {
+        init_test_logging();
+        let test_name = "log_error_works_fine_for_externally_triggered_scanning";
+        let input = vec![
+            (
+                StartScanError::ScanAlreadyRunning {
+                    pertinent_scanner: ScanType::Payables,
+                    started_at: SystemTime::now()
+                },
+                format!("INFO: {test_name}: Payables scan was already initiated at" /*TODO suboptimal */)
+            ),
+            (
+                StartScanError::ManualTriggerError(MTError::AutomaticScanConflict),
+                format!("WARN: {test_name}: User requested Payables scan was denied. Automatic mode prevents manual triggers.")
+            ),
+            (
+                StartScanError::ManualTriggerError(MTError::UnnecessaryRequest {
+                    hint_opt: Some("Wise words".to_string())
+                }),
+                format!("INFO: {test_name}: User requested Payables scan was denied expecting zero findings. Wise words")
+            ),
+            (
+                StartScanError::ManualTriggerError(MTError::UnnecessaryRequest {
+                    hint_opt: None}
+                ),
+                format!("INFO: {test_name}: User requested Payables scan was denied expecting zero findings.")
+            ),
+            (
+                StartScanError::CalledFromNullScanner,
+                format!("WARN: {test_name}: Called from NullScanner, not the Payables scanner.")
+            ),
+            (
+                StartScanError::NoConsumingWalletFound,
+                format!("WARN: {test_name}: Cannot initiate Payables scan because no consuming wallet was found.")
+            ),
+            (
+                StartScanError::NothingToProcess,
+                format!("INFO: {test_name}: There was nothing to process during Payables scan.")
+            ),
+        ];
+        let logger = Logger::new(test_name);
+        let test_log_handler = TestLogHandler::new();
+
+        input.into_iter().for_each(|(err, expected_log_msg)| {
+            err.log_error(&logger, ScanType::Payables, true);
+
+            test_log_handler.exists_log_containing(&expected_log_msg);
+        });
+    }
+
+    fn make_dull_subject() -> Scanners {
+        Scanners {
+            payable: Box::new(NullScanner::new()),
+            aware_of_unresolved_pending_payable: false,
+            initial_pending_payable_scan: false,
+            pending_payable: Box::new(NullScanner::new()),
+            receivable: Box::new(NullScanner::new()),
+        }
+    }
+
+    // Concatenated hours, minutes, seconds and milliseconds in a single integer
+    #[derive(PartialEq, Debug)]
+    struct PseudoTimestamp {
+        rep: u32,
```

Comment:
If you'd like to try you can find a way to convert `String` to `SystemTime`. Then it would be easier to compare. Furthermore, we can use UNIX timestamp for comparison.